test/runnable/test42.d

```diff
@@ -6167,6 +6167,88 @@ void test11472()
 }
 
 
+/***************************************************/
+// .tupleof for static arrays
+
+void test253()
```

Comment:
This test looks very complex for just testing `.tupleof` on a static array, involving associative arrays, destructors, and generated code. How about splitting it up into simpler chunks?

```D
// test .tupleof on static array
void test253()
{
    // simple test
}

// test that static array .tupleof makes no redundant copies
void test253b()
{    
    // test with copy constructors
}
```

Reply:
Also consider making it a compilable, they're much faster. You can test semantics using CTFE

Reply:
Honestly I feel pretty icky about this test too, but I also wanted to thoroughly ensure that the semantics matched that of structs (largely out of paranoia that implementing the feature was *far* too easy, after referencing the impls for structs/classes and seeing special casing for side effects.) Directly comparing copy/destruction counts seemed like the most straightforward way to do that. Am I just being overly cautious?

Reply:
Good points. I should have run a few more reduction passes on this before pulling the trigger :P
I hope the test is a bit more legible now!